packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx

```diff
@@ -0,0 +1,154 @@
+	const { user } = useUser();
+	const [message, setMessage] = useState<string>();
+	const [claiming, setClaiming] = useState(false);
+	const [claimed, setClaimed] = useState(false);
+
+	const confirmBounty = async () => {
+		if (message && user) {
+			try {
+				setClaiming(true);
+				const res = await axios.patch<void, any, BountyClaimCollection>(
+					`api/bounties/${bounty._id}/claim?customerId=${bounty.customerId}`
+					, {
+						claimedBy: claimedBy(user),
+						submissionNotes: message,
+						status: 'In-Progress',
+						statusHistory: newStatusHistory(bounty.statusHistory as StatusHistoryItem[]),
+					}
+				);
+				if (res.status === 200)	{
+					setClaimed(true);
+					await new Promise(resolve => setTimeout(resolve, 2000)).then(() => {
+						router.push('/' + bounty._id);
+						setClaimed(false);
+					});
+				}
+			} finally {
+				setClaiming(false);
+			}
+		} else {
+			throw new Error('Missing Message');
+		}
+	};
+	
+	return (
+		<>
+	  		{
+				bounty.discordMessageId
+				// If we have discord message info for the bounty, the user is taken into discord to claim
+					? <ClaimDiscord discordMessageId={bounty.discordMessageId} />
+				// else, they must use the web form variant
+			 	: <ClaimWeb onOpen={onOpen} />
+			}
+			<Modal onClose={onClose} isOpen={isOpen} isCentered>
+		  		<ModalOverlay />
+		  		<ModalContent>
+					<ModalHeader>Claim This Bounty</ModalHeader>
+					{
+						claimed &&
+					<Alert status='success'>
+    					<AlertIcon />
+						Bounty Claimed!
+  					</Alert>}
+					<ModalCloseButton />
+					<ModalBody
+						flexDirection="column"
+						justifyContent="space-evenly"
+					>
+						<Flex mb="5">
+					Add a message to the bounty creator, then hit 'confirm' to send and claim the bounty.
+						</Flex>
+						<Textarea placeholder='Send a message' onChange={e => setMessage(e.target.value)} />
+					</ModalBody>
+					<ModalFooter justifyContent="start">
+						<Button transition="background 100ms linear"
+							disabled={claimed || !message}
+							onClick={confirmBounty}
+							isLoading={claiming}
+							loadingText='Submitting'
+							bg={colorMode === 'light' ? 'primary.300' : 'primary.700'}
+						>
+						Claim It
+						</Button>
+			  		<Button ml="3" onClick={onClose}>Close</Button>
+					</ModalFooter>
+				</ModalContent>
+			</Modal>
+		</>
+	);
+};
+
+export const ClaimDiscord = ({ discordMessageId }: { discordMessageId: string }): JSX.Element => {
+	const { customer: { customerId, bountyChannel } } = useContext(CustomerContext);
+	const url = `${baseUrl}/${customerId}/${bountyChannel}/${discordMessageId}`;
+	const { colorMode } = useColorMode();
+	return (
+		<AccessibleLink href={url}>
+			<Button
+				transition="background 100ms linear"
+				bg={colorMode === 'light' ? 'primary.300' : 'primary.700'}
+				my={2} size="sm"
+			>
+				Claim It
+			</Button>
+		</AccessibleLink>
+	);
+
+};
+
+export const useCanClaim = (): boolean => {
```

Comment:
extracted hook to make it easier to test - mocking the useContext object with Chakra creates all sorts of headaches